backend/planty/dashboard/views.py

```diff
@@ -290,3 +295,226 @@ def post(self, request: Request):
         plant.save()
 
         return Response(status=status.HTTP_200_OK)
+
+
+class InstructionsView(APIView):
+    def get(self, request: Request):
+        user: User = request.user
+        my_instructions = Instruction.objects.filter(user=user, public=False)
+
+        my_instructions_json = []
+        for instruction in my_instructions:
+            instruction_json = {
+                'id': str(instruction.id),
+                'name' : instruction.name,
+                'species': instruction.species,
+
+                'watering': instruction.watering,
+                'insolation': instruction.insolation,
+                'fertilizing': instruction.fertilizing,
+            }
+            my_instructions_json.append(instruction_json)
+
+        return Response(my_instructions_json, status=status.HTTP_200_OK)
+
+    def post(self, request: Request):
+        user: User = request.user
+        serializer = InstructionCreateSerializer(data=request.data)
+
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        data = serializer.validated_data
+
+        instruction: Instruction = Instruction.objects.create(
+            id=uuid4(),
+            name=name,
+
+            user=user,
+            species=data['species'],
+
+            watering=data['watering'],
+            insolation=data['insolation'],
+            fertilizing=data['fertilizing']
+        )
+        instruction.save()
+
+        return Response(status=status.HTTP_201_CREATED)
+
+    def put(self, request: Request):
+        user: User = request.user
+        serializer = InstructionSelectSerializer(data=self.kwargs['uuid'])
+
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        data = request.data
+
+        try:
+            instruction: Instruction = Instruction.objects.get(id=kwargs['uuid'])
+        except Model.DoesNotExist:
+            return Response(data={
+                'id': ['Plant with the given ID does not exist']
+            }, status=status.HTTP_404_NOT_FOUND)
+
+        if instruction.user != user:
+            return Response(status=status.HTTP_403_FORBIDDEN)
+
+        if instruction.public:
+            return Response(data={
+                'id:': ['Instruction with this ID is public. Cannot modify public instructions.']
+            }, status=status.HTTP_405_METHOD_NOT_ALLOWED)
```

Comment:
To jest fajne, mi się podoba. Nie wiem co prawda czy 405 jest poprawny kod. Może @Kamil153 ma coś do powiedzenia. Ewentualnie można 409 dać. 403 niby bardziej przez brak autentykacji. 400 nie, bo request jest poprawny. 422 jeszcze można ewentualnie. 405 by implikowało, że nie da się na tym url'u puścić PUT, ale da się.

Reply:
> To jest fajne, mi się podoba. Nie wiem co prawda czy 405 jest poprawny kod. Może @Kamil153 ma coś do powiedzenia. Ewentualnie można 409 dać. 403 niby bardziej przez brak autentykacji. 400 nie, bo request jest poprawny. 422 jeszcze można ewentualnie. 405 by implikowało, że nie da się na tym url'u puścić PUT, ale da się.

Jest git jak przeczytałem troszkę inne source'y